examples/python/gds_whole_slide/benchmark_zarr_write_lz4_via_dask.py

```diff
@@ -0,0 +1,152 @@
+"""
+TODO: Currently LZ4-compressed images are a tiny bit larger than uncompressed.
+Need to look into why this is!
+
+"""
+
+import math
+import os
+
+import cupy as cp
+import kvikio.defaults
+import numpy as np
+from cucim.core.operations.color import image_to_absorbance
+from cupyx.profiler import benchmark
+from tifffile import TiffFile
+
+from demo_implementation import read_tiled, get_n_tiles, cupy_to_zarr
+from lz4_nvcomp import LZ4NVCOMP
+
+
+data_dir = os.environ.get('WHOLE_SLIDE_DATA_DIR', os.path.dirname('__file__'))
+fname = os.path.join(data_dir, 'resize.tiff')
+if not os.path.exists(fname):
+    raise RuntimeError(f"Could not find data file: {fname}")
+
+# make sure we are not in compatibility mode to ensure cuFile is being used
+# (when compat_mode() is True, POSIX will be used instead of libcufile.so)
+kvikio.defaults.compat_mode_reset(False)
+assert not kvikio.defaults.compat_mode()
+
+# set the number of threads to use
+kvikio.defaults.num_threads_reset(16)
+
+print(f"\t{kvikio.defaults.compat_mode() = }")
+print(f"\t{kvikio.defaults.get_num_threads() = }")
+print(f"\tkvikio task size = {kvikio.defaults.task_size()/1024**2} MB")
+
+
+n_buffer = 128
+max_duration = 8
+level = 1
+
+with TiffFile(fname) as tif:
+    page = tif.pages[level]
+    page_shape = page.shape
+    tile_shape = (page.tilelength, page.tilewidth, page.samplesperpixel)
+    total_tiles = math.prod(get_n_tiles(page))
+print(f"Resolution level {level}\n")
+print(f"\tshape: {page_shape}")
+print(f"\tstored as {total_tiles} tiles of shape {tile_shape}")
+
+
+# read the uint8 TIFF
+kwargs = dict(levels=[level], backend='kvikio-pread', n_buffer=n_buffer)
+image_gpu = read_tiled(fname, **kwargs)[0]
+
+# benchmark writing these CuPy outputs to Zarr with various chunk sizes
+# Note: nvcomp only supports integer and unsigned dtypes.
+# https://github.com/rapidsai/kvikio/blob/b0c6cedf43d1bc240c3ef1b38ebb9d89574a08ee/python/kvikio/nvcomp.py#L12-L21  # noqa
+
+dtypes = ['uint16']
+chunk_shapes = [(512, 512, 3), (1024, 1024, 3), (2048, 2048, 3), (4096, 4096, 3)]
+backend = 'dask'
+compressors = [None, LZ4NVCOMP()]
+kvikio.defaults.num_threads_reset(16)
+write_time_means = np.zeros(((len(dtypes), len(chunk_shapes), len(compressors), 2)), dtype=float)
+write_time_stds = np.zeros_like(write_time_means)
+for i, dtype in enumerate(dtypes):
+    if dtype == 'uint8':
+        img = image_gpu
+        assert img.dtype == cp.uint8
+    else:
+        img = image_gpu.astype(dtype)
+    else:
+        raise NotImplementedError("only testing for uint8 and uint16")
```

Comment:
This double `else` looks odd. What was intended?

Looking at the error message would have thought this

```suggestion
    if dtype == 'uint8':
        img = image_gpu
        assert img.dtype == cp.uint8
    if dtype == 'uint8':
        img = image_gpu
        assert img.dtype == cp.uint8
    elif dtype == 'uint16':
        img = image_gpu
        assert img.dtype == cp.uint16
        img = image_gpu.astype(dtype)
    else:
        raise NotImplementedError("only testing for uint8 and uint16")
```

Though maybe the error is simply not needed?

```suggestion
    if dtype == 'uint8':
        img = image_gpu
        assert img.dtype == cp.uint8
    else:
        img = image_gpu.astype(dtype)
```

Reply:
cc @bdice

Reply:
For context, I was trying to update #474 which reinstates some style checks that were disabled during the GitHub Actions transition. The Python code fails to parse because it has two else: blocks.

Reply:
Thoughts @grlee77?

Reply:
Creating issue ( https://github.com/rapidsai/cucim/issues/484 ) to track